### ISSUE TITLE
Insecure pickle in case of multiple devs works with the same database

### ISSUE THREAD

**Issue by Proger666** [web2py/web2py#1868]
Web2py have database cash in /databases folder so it’s leads to a problem when several developers connects to database even without changing anything inside database structure.
In order to fix this problem one needs to use fake_migrate_all an drecreate local database cash.

I think it will be much mor effective to sync with database from time to time in order to seamlessly update local files.

**Comment by masia5** [comment]
Share the databases/* files with the other devs, for example with git or local network. Mind that  issue regards https://github.com/web2py/pydal close this one and perhaps open a new one there

**Comment by Proger666** [comment]
Yes, this is a workaround but not really viable solution for production environment when several people connect to dB and modify some parts. Workflow became very clumpy and it’s can be fixed via small fix. For example one of the solution can be in creating cache table in a database and then  systematically local cache updating.

**Comment by leonelcamara** [comment]
I don't understand what's happening here, you have several people changing the .table files on the production environment? What do you mean cache in databases folder?

**Comment by Proger666** [comment]
There are some files in local /database folder,  like a local cache right ? This files constantly changes over time and that leads to errors such as “insecure string pickle” when several ppl use local Web2py process but connected to production database.

**Comment by dartg** [comment]
When multiple web2py applications shares the same database, all of them except one should disable migrations (`migrate_enabled=False`). Otherwise web2py will apply the same migrations as many times as applications sharing the database, completely messing up your schema.

**Comment by leonelcamara** [comment]
@Proger666 what dartg said. Those files are not a "cache" those files represent how the tables were defined the last time web2py touched the database with migration enabled. Web2py uses those files for automatic migrations when you change your tables.  
  
The databases folder should not be shared by all developers and should not be under version control, least of all the databases folder of the application in production. Each dev should have it's own local database he uses for tests.

**Comment by Proger666** [comment]
Yes, this is a good way to do it when everyone have local test that is synced to prod. But what the reason not to store this files in DB?
For example check if dB with this file ps exists - sync, if not migrate.

**Comment by Proger666** [comment]
Anyway this error should be redesigned because it’s tells nothing about root cause of the problem.

**Comment by BuhtigithuB** [comment]
But maybe the whole idea of using migrate=True in production is wrong, I
mean you don't want to apply an hot fix that change your database and let
web2py handle that on the live production environnement...
